src/main/java/javaserver/ResponseBuilderFactory.java

```diff
@@ -0,0 +1,58 @@
+package javaserver;
+
+import javaserver.RequestHandlers.*;
+import javaserver.ResponseBuilders.*;
+
+public class ResponseBuilderFactory {
+
+	public static ResponseBuilder createResponder(Request request) {
+		String requestURI = getURI(request);
+		ResponseBuilder responseBuilder;
+		String requestHandlerType = Routes.acceptableRoutes.get(requestURI);
+
+		if (requestHandlerType == null) {
```

Comment:
It's a little strange that a request that isn't supported is being sent to a default handler. I would think that you'd return something like an ErrorResponseBuilder here that only responds with 404s since you know it's not a defined route at this point. In that way, you could get rid of the notion of a default handler all together and change ResponseBuilder to an interface that all variations of it implement.